lib/aquifer.api.js

```diff
@@ -154,6 +184,95 @@ class Aquifer {
       resolve();
     })
   }
+
+  /**
+   * Get initial project configuration.
+   * @returns {object} promise object.
+   */
+  getConfig() {
+    return new Promise((resolve, reject) => {
+      this.initialConfig = {};
+
+      // If this project is initialized, load the JSON path.
+      if (this.initialized) {
+        // Calculate paths to json, and source directory.
+        let jsonPath = path.join(this.projectDir, 'aquifer.json');
+        let localJsonPath = path.join(this.projectDir, 'aquifer.local.json');
+
+        this.initialConfig = jsonFile.readFileSync(jsonPath);
+
+        // Extend config with aquifer.local.json.
+        if (fs.existsSync(localJsonPath)) {
+          this.initialConfig = _.defaultsDeep(jsonFile.readFileSync(localJsonPath), this.initialConfig);
+        }
+      }
+
+      resolve();
+    })
+  }
+
+  /**
+   * Redirects execution to the proper version of aquifer.
+   * @returns {object} promise object.
+   */
+  switchVersion() {
+    return new Promise((resolve, reject) => {
+      // No need to switch aquifer versions if we don't have an aquifer project, yet.
+      if (!this.initialized) {
+        resolve();
+        return;
+      }
+
+      if (!this.initialConfig.hasOwnProperty('version')) {
+        this.console.log('You have not specified an aquifer version in aquifer.json. Using the default installed version: ' + this.version, 'warning');
+        resolve();
+        return;
+      }
+
+      if (this.initialConfig.version === this.version) {
+        resolve();
+        return;
+      }
+
+      let source = this.initialConfig.source || 'aquifer@' + this.initialConfig.version;
+
+      let newAquifer = new this.api.npm(this, 'aquifer', source);
+
+      if (newAquifer.installed) {
+        let newAquiferDir = fs.realpathSync(path.join(newAquifer.path));
+        let aquiferDir = fs.realpathSync(path.join(path.dirname(module.parent.filename), '..'));
+
+        // Since the child aquifer module is installed, if the child module and
+        // this current instance of aquifer have the same directory, then we are
+        // running a chained instance of aquifer. If that chained instance
+        // doesn't have a matching version property, we have a problem.
+        if (aquiferDir === newAquiferDir && this.initialConfig.version !== this.version) {
+          reject('You have specified a version of aquifer that does not match the version that was installed. You probably have misconfigured source or version properties in your aquifer.json file.');
+          return;
+        }
+      }
+
+      // Let the user know we'll be installing the specified version of aquifer now.
+      if (!newAquifer.installed) {
+        this.console.log('Aquifer ' + this.initialConfig.version + ' is not installed. Installing now...');
+      }
```

Comment:
actually techincally speaking... here you don't even need an a check... you just need to console.log because everything below happens when an install is eminent. What I don't get though is you check to see if the version is installed... and if it is... and it's the same version... you go ahead and install anyway again.